Phantomjs 1.9.16 does not install on Ubuntu 14.04 running on Vagrant shared folder

The issue, as originally reported in #279 is still an issue on Ubuntu 14.04 running on a Vagrant shared folder.

```
$ npm install phantomjs
\
> phantomjs@1.9.16 install /vagrant_host/ynab_shared_library/node_modules/phantomjs
> node install.js

Downloading https://bitbucket.org/ariya/phantomjs/downloads/phantomjs-1.9.8-linux-x86_64.tar.bz2
Saving to /tmp/phantomjs/phantomjs-1.9.8-linux-x86_64.tar.bz2
Receiving...
  [=======================================-] 97% 0.0s
Received 12854K total.
Extracting tar contents (via spawned process)
Removing /vagrant_host/ynab_shared_library/node_modules/phantomjs/lib/phantom
Copying extracted folder /tmp/phantomjs/phantomjs-1.9.8-linux-x86_64.tar.bz2-extract-1428441311626/phantomjs-1.9.8-linux-x86_64 -> /vagrant_host/ynab_shared_library/node_modules/ph
antomjs/lib/phantom
Writing location.js file
Phantom installation failed { [Error: ENOTDIR, not a directory '/vagrant_host/ynab_shared_library/node_modules/phantomjs/lib/phantom/bin/phantomjs']
  errno: 27,
  code: 'ENOTDIR',
  path: '/vagrant_host/ynab_shared_library/node_modules/phantomjs/lib/phantom/bin/phantomjs',
  syscall: 'chmod' } Error: ENOTDIR, not a directory '/vagrant_host/ynab_shared_library/node_modules/phantomjs/lib/phantom/bin/phantomjs'
    at Object.fs.chmodSync (evalmachine.<anonymous>:832:18)
    at Object.chmodSync (/vagrant_host/ynab_shared_library/node_modules/phantomjs/node_modules/fs-extra/node_modules/graceful-fs/polyfills.js:141:17)
    at Promise.validExit [as _successFn] (/vagrant_host/ynab_shared_library/node_modules/phantomjs/install.js:145:8)
    at Promise._call (/vagrant_host/ynab_shared_library/node_modules/phantomjs/node_modules/kew/kew.js:373:13)
    at Promise._withInput (/vagrant_host/ynab_shared_library/node_modules/phantomjs/node_modules/kew/kew.js:333:25)
    at Promise.resolve (/vagrant_host/ynab_shared_library/node_modules/phantomjs/node_modules/kew/kew.js:105:27)
    at resolver (/vagrant_host/ynab_shared_library/node_modules/phantomjs/node_modules/kew/kew.js:409:17)
    at next (/vagrant_host/ynab_shared_library/node_modules/phantomjs/node_modules/fs-extra/node_modules/rimraf/rimraf.js:70:7)
    at CB (/vagrant_host/ynab_shared_library/node_modules/phantomjs/node_modules/fs-extra/node_modules/rimraf/rimraf.js:106:9)
    at Object.oncomplete (fs.js:107:15)
npm ERR! Linux 3.2.0-23-generic
npm ERR! argv "node" "/usr/bin/npm" "install" "phantomjs"
npm ERR! node v0.10.32
npm ERR! npm  v2.7.5
npm ERR! code ELIFECYCLE

npm ERR! phantomjs@1.9.16 install: `node install.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the phantomjs@1.9.16 install script 'node install.js'.
npm ERR! This is most likely a problem with the phantomjs package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node install.js
npm ERR! You can get their info via:
npm ERR!     npm owner ls phantomjs
npm ERR! There is likely additional logging output above.
npm ERR! Linux 3.2.0-23-generic
npm ERR! argv "node" "/usr/bin/npm" "install" "phantomjs"
npm ERR! node v0.10.32
npm ERR! npm  v2.7.5
npm ERR! path npm-debug.log.d96adeda5efd8bee9b90041799f1e1c1
npm ERR! code ETXTBSY
npm ERR! errno 62

npm ERR! ETXTBSY, rename 'npm-debug.log.d96adeda5efd8bee9b90041799f1e1c1'
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     /vagrant_host/ynab_shared_library/npm-debug.log
```

fwiw, I tried installing phantomjs on a vagrant shared folder on ubuntu,
and it worked fine for me.

On Tue, May 19, 2015 at 8:59 PM, Nick Barrett <notifications@github.com>
wrote:

> The issue, as originally reported in #279
> <https://github.com/Medium/phantomjs/issues/279> is still an issue on
> Ubuntu 14.04 running on a Vagrant shared folder.
>
> $ npm install phantomjs
> \
> > phantomjs@1.9.16 install /vagrant_host/ynab_shared_library/node_modules/phantomjs
> > node install.js
>
> Downloading https://bitbucket.org/ariya/phantomjs/downloads/phantomjs-1.9.8-linux-x86_64.tar.bz2
> Saving to /tmp/phantomjs/phantomjs-1.9.8-linux-x86_64.tar.bz2
> Receiving...
>   [=======================================-] 97% 0.0s
> Received 12854K total.
> Extracting tar contents (via spawned process)
> Removing /vagrant_host/ynab_shared_library/node_modules/phantomjs/lib/phantom
> Copying extracted folder /tmp/phantomjs/phantomjs-1.9.8-linux-x86_64.tar.bz2-extract-1428441311626/phantomjs-1.9.8-linux-x86_64 -> /vagrant_host/ynab_shared_library/node_modules/ph
> antomjs/lib/phantom
> Writing location.js file
> Phantom installation failed { [Error: ENOTDIR, not a directory '/vagrant_host/ynab_shared_library/node_modules/phantomjs/lib/phantom/bin/phantomjs']
>   errno: 27,
>   code: 'ENOTDIR',
>   path: '/vagrant_host/ynab_shared_library/node_modules/phantomjs/lib/phantom/bin/phantomjs',
>   syscall: 'chmod' } Error: ENOTDIR, not a directory '/vagrant_host/ynab_shared_library/node_modules/phantomjs/lib/phantom/bin/phantomjs'
>     at Object.fs.chmodSync (evalmachine.<anonymous>:832:18)
>     at Object.chmodSync (/vagrant_host/ynab_shared_library/node_modules/phantomjs/node_modules/fs-extra/node_modules/graceful-fs/polyfills.js:141:17)
>     at Promise.validExit [as _successFn] (/vagrant_host/ynab_shared_library/node_modules/phantomjs/install.js:145:8)
>     at Promise._call (/vagrant_host/ynab_shared_library/node_modules/phantomjs/node_modules/kew/kew.js:373:13)
>     at Promise._withInput (/vagrant_host/ynab_shared_library/node_modules/phantomjs/node_modules/kew/kew.js:333:25)
>     at Promise.resolve (/vagrant_host/ynab_shared_library/node_modules/phantomjs/node_modules/kew/kew.js:105:27)
>     at resolver (/vagrant_host/ynab_shared_library/node_modules/phantomjs/node_modules/kew/kew.js:409:17)
>     at next (/vagrant_host/ynab_shared_library/node_modules/phantomjs/node_modules/fs-extra/node_modules/rimraf/rimraf.js:70:7)
>     at CB (/vagrant_host/ynab_shared_library/node_modules/phantomjs/node_modules/fs-extra/node_modules/rimraf/rimraf.js:106:9)
>     at Object.oncomplete (fs.js:107:15)
> npm ERR! Linux 3.2.0-23-generic
> npm ERR! argv "node" "/usr/bin/npm" "install" "phantomjs"
> npm ERR! node v0.10.32
> npm ERR! npm  v2.7.5
> npm ERR! code ELIFECYCLE
>
> npm ERR! phantomjs@1.9.16 install: `node install.js`
> npm ERR! Exit status 1
> npm ERR!
> npm ERR! Failed at the phantomjs@1.9.16 install script 'node install.js'.
> npm ERR! This is most likely a problem with the phantomjs package,
> npm ERR! not with npm itself.
> npm ERR! Tell the author that this fails on your system:
> npm ERR!     node install.js
> npm ERR! You can get their info via:
> npm ERR!     npm owner ls phantomjs
> npm ERR! There is likely additional logging output above.
> npm ERR! Linux 3.2.0-23-generic
> npm ERR! argv "node" "/usr/bin/npm" "install" "phantomjs"
> npm ERR! node v0.10.32
> npm ERR! npm  v2.7.5
> npm ERR! path npm-debug.log.d96adeda5efd8bee9b90041799f1e1c1
> npm ERR! code ETXTBSY
> npm ERR! errno 62
>
> npm ERR! ETXTBSY, rename 'npm-debug.log.d96adeda5efd8bee9b90041799f1e1c1'
> npm ERR!
> npm ERR! If you need help, you may report this error at:
> npm ERR!     <https://github.com/npm/npm/issues>
>
> npm ERR! Please include the following file with any support request:
> npm ERR!     /vagrant_host/ynab_shared_library/npm-debug.log
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Medium/phantomjs/issues/341>.
>

C:\Users\jlopesde\heroku\desolate-taiga-1397>npm install

> phantomjs@1.9.17 install C:\Users\jlopesde\heroku\desolate-taiga-1397\node_mod
ules\phantomjs
> node install.js

Download already available at C:\Users\jlopesde\AppData\Local\Temp\phantomjs\pha
ntomjs-1.9.8-windows.zip
Extracting zip contents
Removing C:\Users\jlopesde\heroku\desolate-taiga-1397\node_modules\phantomjs\lib
\phantom
Phantom installation failed { [Error: EPERM, unlink 'C:\Users\jlopesde\heroku\de
solate-taiga-1397\node_modules\phantomjs\lib\phantom\phantomjs.exe']
  errno: 50,
  code: 'EPERM',
  path: 'C:\\Users\\jlopesde\\heroku\\desolate-taiga-1397\\node_modules\\phantom
js\\lib\\phantom\\phantomjs.exe' } Error: EPERM, unlink 'C:\Users\jlopesde\herok
u\desolate-taiga-1397\node_modules\phantomjs\lib\phantom\phantomjs.exe'

npm ERR! phantomjs@1.9.17 install: `node install.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the phantomjs@1.9.17 install script.
npm ERR! This is most likely a problem with the phantomjs package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node install.js
npm ERR! You can get their info via:
npm ERR!     npm owner ls phantomjs
npm ERR! There is likely additional logging output above.
npm ERR! System Windows_NT 6.1.7601
npm ERR! command "C:\\Program Files\\nodejs\\\\node.exe" "C:\\Program Files\\nod
ejs\\node_modules\\npm\\bin\\npm-cli.js" "install"
npm ERR! cwd C:\Users\jlopesde\heroku\desolate-taiga-1397
npm ERR! node -v v0.10.33
npm ERR! npm -v 1.4.28
npm ERR! code ELIFECYCLE
npm ERR!
npm ERR! Additional logging details can be found in:
npm ERR!     C:\Users\jlopesde\heroku\desolate-taiga-1397\npm-debug.log
npm ERR! not ok code 0

So coming back to this issue I recently installed 1.9.17 on the same platform as described in my OP.
This was the error again:

```
> phantomjs@1.9.17 install /var/www/node_modules/phantomjs
> node install.js

Downloading https://bitbucket.org/ariya/phantomjs/downloads/phantomjs-1.9.8-linux-x86_64.tar.bz2
Saving to /tmp/phantomjs/phantomjs-1.9.8-linux-x86_64.tar.bz2
Receiving...
  [=======================================-] 96% 0.0s
Received 12854K total.
Extracting tar contents (via spawned process)
Removing /var/www/node_modules/phantomjs/lib/phantom
Copying extracted folder /tmp/phantomjs/phantomjs-1.9.8-linux-x86_64.tar.bz2-extract-1434938433147/phantomjs-1.9.8-linux-x86_64 -> /var/www/node_modules/phantomjs/lib/phantom
Phantom installation failed { [Error: ENOTDIR, not a directory '/var/www/node_modules/phantomjs/lib/phantom/bin/phantomjs']
  errno: 27,
  code: 'ENOTDIR',
  path: '/var/www/node_modules/phantomjs/lib/phantom/bin/phantomjs',
  syscall: 'chmod' } Error: ENOTDIR, not a directory '/var/www/node_modules/phantomjs/lib/phantom/bin/phantomjs'
    at Object.fs.chmodSync (evalmachine.<anonymous>:833:18)
    at Object.chmodSync (/var/www/node_modules/phantomjs/node_modules/fs-extra/node_modules/graceful-fs/polyfills.js:141:17)
    at Promise.validExit [as _successFn] (/var/www/node_modules/phantomjs/install.js:144:10)
    at Promise._call (/var/www/node_modules/phantomjs/node_modules/kew/kew.js:373:13)
    at Promise._withInput (/var/www/node_modules/phantomjs/node_modules/kew/kew.js:333:25)
    at Promise.resolve (/var/www/node_modules/phantomjs/node_modules/kew/kew.js:105:27)
    at resolver (/var/www/node_modules/phantomjs/node_modules/kew/kew.js:409:17)
    at next (/var/www/node_modules/phantomjs/node_modules/fs-extra/node_modules/rimraf/rimraf.js:72:7)
    at CB (/var/www/node_modules/phantomjs/node_modules/fs-extra/node_modules/rimraf/rimraf.js:108:9)
    at Object.oncomplete (fs.js:108:15)

npm ERR! phantomjs@1.9.17 install: `node install.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the phantomjs@1.9.17 install script.
npm ERR! This is most likely a problem with the phantomjs package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node install.js
npm ERR! You can get their info via:
npm ERR!     npm owner ls phantomjs
npm ERR! There is likely additional logging output above.
npm ERR! System Linux 3.16.0-41-generic
npm ERR! command "/usr/bin/node" "/usr/bin/npm" "install"
npm ERR! cwd /var/www
npm ERR! node -v v0.10.40
npm ERR! npm -v 1.4.28
npm ERR! code ELIFECYCLE
npm ERR! not ok code 0
```

--- a/package.json
+++ b/package.json
@@ -35,7 +35,7 @@
   },
   "dependencies": {
     "adm-zip": "0.4.4",
-    "fs-extra": "~0.18",
+    "fs-extra": "~0.23.1",
     "kew": "0.4.0",
     "npmconf": "2.1.1",
     "progress": "1.1.8",

